src/S3/AmbiguousSuccessParser.php

```diff
@@ -0,0 +1,41 @@
+<?php
+namespace Aws\S3;
+
+use Aws\Api\Parser\AbstractParser;
+use Aws\CommandInterface;
+use Psr\Http\Message\ResponseInterface;
+
+class AmbiguousSuccessParser extends AbstractParser
+{
+    private static $ambiguousSuccesses = [
+        'UploadPartCopy' => true,
+        'CopyObject' => true,
+        'CompleteMultipartUpload' => true,
+    ];
+
+    /** @var callable */
+    private $parser;
+
+    public function __construct(callable $parser)
+    {
+        $this->parser = $parser;
+    }
+
+    public function __invoke(
+        CommandInterface $command,
+        ResponseInterface $response
+    ) {
+        $fn = $this->parser;
```

Comment:
They are separate errors, having nothing in common besides their being error responses with a 200 status code, so they could be caught by separate parsers.